File: cppversion/38. count and say.cpp
#include <iostream>
#include <string>

std::string countAndSay(int n)
{
    if (n == 0)
        return "";
    std::string res = "1";
    while (--n)
    {
        std::string cur = "";
        for (int i = 0; i < res.size(); i++)
        {
            int count = 1;

            while (i + 1 < res.size() && (res[i] == res[i + 1]))
            {
                count++;
                i++;
            }
            cur += std::to_string(count) + res[i];
        }
        res = cur;
    }
    return res;
}

int main()
{
    std::string res;
    res = countAndSay(10);
    std::cout << res << std::endl;
    std::cin.get();
}

File: pyversion/38. count and say.py
def countAndSay(n):
    if n == 0:
        return ''
    res = '1'
    n = n - 1
    while (n):
        cur = ''
        i = 0
        while (i < len(res)):
            count = 1
            while (i + 1 < len(res) and res[i] == res[i + 1]):
                count += 1
                i += 1
            cur = cur + str(count) + res[i]
            i += 1
        res = cur
        n -= 1
    return res


if __name__ == '__main__':
    print(countAndSay(5))